### PR TITLE
Respect textwidth for nixfmt

### PR DESCRIPTION
**Pull request by lstrojny**
`nixfmt` takes a `--width` argument as well, so let's `textwidth` there.

### CODE REVIEW

File: plugin/defaults.vim [hunk 1]
```diff
@@ -690,7 +690,7 @@ endif
 
 " Nix
 if !exists('g:formatdef_nix_format')
-    let g:formatdef_nix_format = '"nixfmt"'
+    let g:formatdef_nix_format = '"nixfmt --width="&textwidth'
```
Comment (chtenb) [inline plugin/defaults.vim:693]: I think there is a string concatenation operator missing here (`.`)

Comment (lstrojny) [reply plugin/defaults.vim]: Indeed, fixed

**Comment by chtenb** [comment]
Does nixfmt behave correctly when textwidth is 0? If nixfmt ignores the argument in that case, the change looks good to me.

**Comment by lstrojny** [comment]
> Does nixfmt behave correctly when textwidth is 0? If nixfmt ignores the argument in that case, the change looks good to me.

Of course it doesn’t 😄 

Made the setting conditional

**Pull request merged**